src/uwtools/config/jinja2.py

```diff
@@ -30,19 +22,44 @@ class J2Template:
     Reads Jinja2 templates from files or strings, and renders them using the user-provided values.
     """
 
-    def __init__(self, values: dict, template_source: Union[str, Path]) -> None:
+    def __init__(
+        self,
+        values: dict,
+        template_source: Optional[Union[str, Path]] = None,
+        searchpath: Optional[str] = None,
+    ) -> None:
         """
         :param values: Values needed to render the provided template.
         :param template_source: Jinja2 string or template file path (None => read stdin).
+        :param searchpath: Colon-separated paths to search for extra templates.
         :raises: RuntimeError: If neither a template file or path is provided.
         """
         self._values = values
-        self._template = (
-            self._load_string(template_source)
-            if isinstance(template_source, str)
-            else self._load_file(template_source)
-        )
         self._template_source = template_source
+        self._j2env = Environment(
+            loader=FileSystemLoader(
+                searchpath=searchpath.split(":")
+                if searchpath
+                else self._template_source.parent
+                if isinstance(self._template_source, Path)
+                else []
```

Comment:
Note the precedence for the search path:
1. Use the explicitly specified search path(s) if provided;
2. Use the directory containing the input template if possible;
3. Use no search path, as none can be defined.